File: September 2026/2026-09-12/154-find-min-in-sorted-array-ii.py
# 154-find-min-in-sorted-array-ii.cpp;binary-search;self;leetcode

class Solution:
    def findMin(self, nums):
        left = 0
        right = len(nums) - 1

        while left < right:
            mid = left + (right - left) // 2

            if nums[mid] < nums[right]:
                right = mid
            elif nums[mid] > nums[right]:
                left = mid + 1
            else:
                right -= 1

        return nums[left]

File: September 2026/2026-09-12/154-find-min-in-sorted-array-ii.cpp
// 154-find-min-in-sorted-array-ii.cpp;binary-search;self;leetcode

class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0, right = nums.size() - 1;
        while(left < right ) {
            int mid = left + (right - left) / 2;
            if(nums[mid] < nums[right]) right = mid;
            else if(nums[mid] > nums[right]) left = mid + 1;
            else right--;
        }

        return nums[left];
    }
};